api/autorest-config.yaml

```
@@ -0,0 +1,7 @@
+input-file: redhatopenshift/resource-manager/Microsoft.RedHatOpenshift/preview/2024-06-10-preview/openapi.json
+go:
+  namespace: redhatopenshift
+  output-folder: generated/hcp
+  module-version: "0.0.1"
+  module: "hcp"
+  azure-arm: true
```

I played around with the options and have some suggestions to better integrate this into the existing source tree.

Note: I disabled fakes temporarily because I'm getting an error I haven't figured out how to resolve:
> fatal   | Error: unable to determine containing module for fakes. specify either the module or containing-module switch

Can't find any documentation about a "containing-module" switch.

```suggestion
  namespace: redhatopenshift
  project-folder: ../internal
  output-folder: $(project-folder)/api/v20240610preview
  file-prefix: generated
  module-version: "0.0.1"
  azure-arm: true
  generate-fakes: false
```